### DISCUSSION TITLE
Practical implications of large number of queues

### DISCUSSION THREAD

**Discussion by chriszarate** [taskforcesh/bullmq#2520] [Q&A] [answered]
The [docs](https://docs.bullmq.io/bull/quick-guide#queues-performance) say:

> Queues are relative cheap, so if you need many of them just create new ones with different names. However having too many queues can become unmanageable. Up to a dozen is normally ok.
> 
> Also keep in mind that every queue instance will require new redis connections, check how to [reuse connections](https://github.com/OptimalBits/bull/blob/master/PATTERNS.md#reusing-redis-connections) or you can also use [named processors](https://github.com/OptimalBits/bull/blob/master/REFERENCE.md#queueprocess) to achieve a similar result.

The heading ("Queues performance") implies that this is a performance limitation. Aside from Redis connection limits, what other performance implications are there for exceeding "up to a dozen" queues? If not performance, what else is implied by "unmanageable"?

I'm hoping to quantify whether any of these limitations can be addressed with resourcing at the app or Redis level, or if there are universal limits. Thanks!

**Comment by manast** [comment, marked as answer]
There are no performance overhead for having more queues, but if you have many of them it is just difficult to handle, as to get visibility on all your queues, how they are processing the jobs, checking for failed jobs, monitoring, etc.

**Reply by chriszarate** [reply]
Thanks for the additional info!
